cmd/ethrex/ethrex.rs

```diff
@@ -401,13 +400,26 @@ fn parse_socket_addr(addr: &str, port: &str) -> io::Result<SocketAddr> {
         ))
 }
 
-fn sync_mode(matches: &clap::ArgMatches) -> SyncMode {
+fn sync_mode(matches: &clap::ArgMatches, chain_config: &ChainConfig) -> SyncMode {
     let syncmode = matches.get_one::<String>("syncmode");
-    match syncmode {
+    let sync_mode = match syncmode {
         Some(mode) if mode == "full" => SyncMode::Full,
         Some(mode) if mode == "snap" => SyncMode::Snap,
+        None => SyncMode::Full,
         other => panic!("Invalid syncmode {:?} expected either snap or full", other),
+    };
+    if sync_mode == SyncMode::Full
+        && (chain_config
+            .terminal_total_difficulty
```

Comment:
Would be better to use another metric that is not total difficulty. 

The logic needs to be a bit more nuanced:
- If the user has selected full sync AND the chain has PoW AND the last block is before the merge => then warn (only if the user explicitly selected full sync) and switch to snap sync

Reply:
This can be tackled later, it is low priority.